LockboxXCUITests/LockBoxScreenGraph.swift

```diff
@@ -0,0 +1,159 @@
+        screenState.gesture(forAction: Action.FxATapOnSignInButton) { userState in
+            app.webViews.buttons["Sign in"].tap()
+        }
+
+        screenState.gesture(forAction: Action.FxALogInSuccessfully, transitionTo: LockboxMainPage) { userState in
+            app.webViews.buttons["Sign in"].tap()
+            userState.fxaUsername = userState.fxaUsername!
+            userState.fxaPassword = userState.fxaPassword!
+        }
+        screenState.tap(app.webViews.links["Create an account"], to: FxCreateAccount)
+
+        screenState.noop(to: LockboxMainPage)
+    }
+
+    map.addScreenState(FxCreateAccount) { screenState in
+        screenState.backAction = navigationControllerBackAction
+    }
+
+    map.addScreenState(LockboxMainPage) { screenState in
+        screenState.tap(app.buttons["preferences"], to: SettingsMenu)
+
+        screenState.gesture(forAction: Action.ChangeEntriesOrder, transitionTo: SortEntriesMenu) { userState in
+            if (app.buttons["A-Z"].exists) {
+                app.buttons["A-Z"].tap()
+            } else {
+                app.buttons["Recent"].tap()
+            }
+        }
+
+        screenState.dismissOnUse = true
+    }
+
+    map.addScreenState(SortEntriesMenu) { screenState in
+        //screenState.backAction = cancelBackAction
+        screenState.noop(to: LockboxMainPage)
+    }
+
+    map.addScreenState(SettingsMenu) { screenState in
+        screenState.tap(app.tables.cells.staticTexts["Open Websites in"], to: OpenSitesInMenu)
+        screenState.tap(app.tables.cells.staticTexts["Account"], to: AccountSettingsMenu)
+        screenState.tap(app.tables.cells.staticTexts["Auto Lock"], to: AutolockSettingsMenu)
+
+        screenState.gesture(forAction: Action.LockNow, transitionTo: LockedScreen) { userState in
+            app.buttons["Lock Now"].tap()
+        }
+        screenState.dismissOnUse = true
+    }
+
+    map.addScreenState(OpenSitesInMenu) { screenState in
+        screenState.backAction = navigationControllerBackAction
+    }
+
+    map.addScreenState(AccountSettingsMenu) { screenState in
+        screenState.backAction = navigationControllerBackAction
+        screenState.dismissOnUse = true
+
+        screenState.gesture(forAction: Action.DisconnectFirefoxLockbox, transitionTo: WelcomeScreen) { userState in
+            app.buttons["Disconnect Firefox Lockbox"].tap()
+            app.buttons["Disconnect"].tap()
+        }
+        screenState.gesture(forAction: Action.DisconnectFirefoxLockboxCancel, transitionTo: AccountSettingsMenu) { userState in
+            app.buttons["Disconnect Firefox Lockbox"].tap()
+            app.buttons["Cancel"].tap()
+        }
+
+    }
+
+    map.addScreenState(AutolockSettingsMenu) { screenState in
+        screenState.backAction = navigationControllerBackAction
+        screenState.dismissOnUse = true
+    }
+
+    map.addScreenState(LockedScreen) { screenState in
+        screenState.backAction = navigationControllerBackAction
```

Comment:
If the locked screen has a back action, then this would be a bug. 

I suggest you add a couple of properties to the user state.

To unlock, you need to be using biometrics or a passcode.

Reply:
There is not a back action, no bug there, I should not have added that back action there, just added the LockedScreen to have that for the future and copy/paste from another screen state